citation/management/commands/export_data.py

```diff
@@ -20,35 +19,62 @@ def add_arguments(self, parser):
+        return output
+
     def handle(self, *args, **options):
         logger.debug("Starting to export data. Hang tight, this may take a while.")
-        header = ["Status", "Publication Title", "Code Url", "Publication Year", "Journal Title", "Primary Author",
-                  "Docs"]
-        max_platforms = self.find_max('platforms')
-        max_sponsors = self.find_max('sponsors')
-        header.extend(self.get_attribute_headers("Platform", max_platforms))
-        header.extend(self.get_attribute_headers("Sponsor", max_sponsors))
-        publications = Publication.objects.all().prefetch_related('sponsors', 'platforms').select_subclasses()
-
-        filename = options['outfile']
-        with open(filename, 'wb') as csvfile:
-            writer = csv.writer(csvfile, delimiter=',')
-            writer.writerow(header)
-            for pub in publications:
-                row = [pub.status, pub.title, pub.code_archive_url, pub.date_published.year or pub.date_published_text,
-                       str(pub.container), str(pub.creators.first()), str(pub.model_documentation)]
-                row.extend(self.get_attribute_values(pub.platforms.all(), max_platforms))
-                row.extend(self.get_attribute_values(pub.sponsors.all(), max_sponsors))
-                writer.writerow(row)
+        header = ["Count", "Id", "Publication Title", "Abstract", "Short Title", "Zotero Key", "Code Url",
+                  "Date Published Text", "Date Accessed", "Archive", "Archive location", "Library catalog",
+                  "Call number", "Rights", "Extra", "Published Language", "Date Added", "Date Modified",
+                  "Zotero Date Added", "Zotero Date Modified", "Status", "Code Archive url", "Contact Email",
+                  "Author Comments", "Email Sent out", "Added by", "Assign Curator", "Conatct Author Name",
+                  "Resource type", "Is primary", "Journal Id", "doi", "Series text", "Series title", "Series", "Issue",
+                  "Volume", "ISSN", "pages", "Year of Publication", "Journal", "Notes", "Platform List"]
+        all_platforms = Platform.objects.all().values_list("name").order_by("name")
+        all_sponsors = Sponsor.objects.all().values_list("name").order_by("name")
+        header.extend(self.get_attribute_values(all_platforms))
+        header.append("Sponsors List")
+        header.extend(self.get_attribute_values(all_sponsors))
+        publications = Publication.objects.filter(is_primary=True).prefetch_related('sponsors', 'platforms')
+        COUNT = 1
+        writer = options['outfile']
+        writer.writerow(header)
+        for pub in publications:
+            platforms = self.get_attribute_values(list(pub.platforms.all().values_list("name")))
+            sponsors = self.get_attribute_values(list(pub.sponsors.all().values_list("name")))
+            notes = self.get_attribute_values(list(Note.objects.filter(publication=pub.pk).values_list("text")))
```

Comment:
You can retrieve notes more naturally in Django using the reverse related_name, e.g., `pub.note_set.all()`